egs/swbd/s5c/local/nnet3/run_tdnn.sh

```diff
@@ -41,26 +44,52 @@ ali_dir=exp/tri4_ali_nodup$suffix
 
 local/nnet3/run_ivector_common.sh --stage $stage \
 	--speed-perturb $speed_perturb || exit 1;
+
 if [ $stage -le 9 ]; then
+  echo "$0: creating neural net configs";
+
+  # create the config files for nnet initialization
+  python steps/nnet3/tdnn/make_configs.py  \
+    --feat-dir data/${train_set}_hires \
+    --ivector-dir exp/nnet3/ivectors_${train_set} \
+    --ali-dir $ali_dir \
+    --relu-dim 1024 \
+    --splice-indexes "-2,-1,0,1,2 -1,2 -3,3 -7,2 0"  \
+    --use-presoftmax-prior-scale true \
+   $dir/configs || exit 1;
+fi
+
+
+
+if [ $stage -le 10 ]; then
   if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $dir/egs/storage ]; then
     utils/create_split_dir.pl \
      /export/b0{3,4,5,6}/$USER/kaldi-data/egs/swbd-$(date +'%m_%d_%H_%M')/s5/$dir/egs/storage $dir/egs/storage
   fi
 
-  steps/nnet3/train_tdnn.sh --stage $train_stage \
-    --num-epochs 2 --num-jobs-initial 3 --num-jobs-final 16 \
-    --splice-indexes "-2,-1,0,1,2 -1,2 -3,3 -7,2 0" \
-    --feat-type raw \
-    --online-ivector-dir exp/nnet3/ivectors_${train_set} \
-    --cmvn-opts "--norm-means=false --norm-vars=false" \
-    --initial-effective-lrate 0.0017 --final-effective-lrate 0.00017 \
-    --cmd "$decode_cmd" \
-    --relu-dim 1024 \
-    data/${train_set}_hires data/lang $ali_dir $dir  || exit 1;
+  steps/nnet3/train_dnn.py --stage=$train_stage \
+    --command="$decode_cmd" \
```

Comment:
I think it's better to use "cmd" as it would then be the same as the shell scripts.